lib/routes/whu/rsgis.ts

```diff
@@ -0,0 +1,285 @@
+        .toArray()
+        .map((item) => parseListLinkDateItem($(item), baseUrl));
+    // 通知公告
+    const tzggList: Array<Post> = $('div.main1 > div.newspaper:nth-child(2) > div.newspaper_list > ul > li')
+        .toArray()
+        .map((item) => parseListLinkDateItem($(item), baseUrl));
+    // 学术动态
+    const xsdtList: Array<Post> = $('div.main3 div.inner > div.newspaper:nth-child(1) > ul.newspaper_list2 > li:nth-child(1) > ul > li')
+        .toArray()
+        .map((item) => parseListLinkDateItem($(item), baseUrl));
+    // 学术进展
+    const xsjzList: Array<Post> = $('div.main3 div.inner > div.newspaper:nth-child(1) > ul.newspaper_list2 > li:nth-child(2) > ul > li')
+        .toArray()
+        .map((item) => parseListLinkDateItem($(item), baseUrl));
+    // 教学动态
+    const jxdtList: Array<Post> = $('div.main3 div.inner > div.newspaper:nth-child(2) > div.newspaper_list2 > ul > li')
+        .toArray()
+        .map((item) => parseListLinkDateItem($(item), baseUrl));
+    // 学工动态
+    const xgdtList: Array<Post> = $('div.main3 div.inner > div.newspaper:nth-child(3) > div.newspaper_list2 > ul > li')
+        .toArray()
+        .map((item) => parseListLinkDateItem($(item), baseUrl));
+    // 组合所有新闻
+    const fullList = await Promise.all(
+        [...xyxwList, ...tzggList, ...xsdtList, ...xsjzList, ...jxdtList, ...xgdtList].map(async (item) => ({
+            ...item,
+            ...(await getDetail(item)),
+        }))
+    );
+    return fullList;
+}
+
+/**
+ * Process non-index types.
+ *
+ * @param type Level 1 type
+ * @param sub Level 2 type
+ */
+async function handlePostList(type: string, sub: string): Promise<Array<DataItem>> {
+    let urlList: Array<{ url: string; base: string }> = [];
+    const category = categoryMap[type];
+    if (sub === 'all') {
+        const subMap = category.sub;
+        urlList = Object.keys(subMap).map((key) => {
+            const subType = subMap[key];
+            return {
+                url: `${baseUrl}/${category.path}/${subType.path}.htm`,
+                base: `${baseUrl}/${category.path}`,
+            };
+        });
+    } else if (sub in category.sub) {
+        urlList.push({
+            url: `${baseUrl}/${category.path}/${category.sub[sub].path}.htm`,
+            base: `${baseUrl}/${category.path}`,
+        });
+    } else {
+        throw new Error('No such sub type.');
+    }
+    const urlPosts = await Promise.all(
+        urlList.map(async (url) => {
+            const response = await ofetch(url.url);
+            const $ = load(response);
+            return $('div.neiinner > div.nav_right > div.right_inner > div.list > ul > li')
+                .toArray()
+                .map((item) => parseListLinkDateItem($(item), url.base));
+        })
+    );
+    const fullList = await Promise.all(
+        urlPosts.flat().map(async (item) => ({
+            ...item,
+            ...(await getDetail(item)),
```

Comment:
Since `item` passed into `getDetail` will be returned as well
https://github.com/DIYgod/RSSHub/blob/4b65c6e4323310df99601a4d8618bf7c3c200fb8/lib/routes/whu/rsgis.ts#L136-L137

You can returned `await getDetail(item)` directly without spreading `item` and `(await getDetail(item))`.